src/vivarium_public_health/dataset_manager/artifact.py

```diff
@@ -31,11 +37,11 @@ def __init__(self, path: str, filter_terms: List[str]=None):
             A set of terms suitable for usage with the ``where`` kwarg for ``pd.read_hdf``
         """
 
+        create_hdf_with_keyspace(path)
         self.path = path
         self._filter_terms = filter_terms
-
         self._cache = {}
-        self._keys = [EntityKey(k) for k in hdf.get_keys(self.path)]
+        self._keys = [EntityKey(k) for k in hdf.load(self.path, EntityKey('metadata.keyspace'), None)]
```

Comment:
I'm thinking that it might be worthwhile to have a small `Keys` class that's responsible for maintaining the relationship between the hdf keyspace and the artifact keyspace.  E.g.:

```
class Keys:

    keyspace_node = EntityKey('metadata.keyspace')

    def __init__(self, artifact_path):
        self.artifact_path = artifact_path
        self._keys = [EntityKey(k) for k in hdf.load(self.path, EntityKey('metadata.keyspace'), None)]

    def append(self, new_key):
        new_key = EntityKey(new_key)
        self._keys.append(new_key)
        hdf.remove(self.artifact_path, self.keyspace_node)
        hdf.write(self.artifact_path, self.keyspace_node, self._keys)

   def remove(self, key):
        similar

    def to_list(self):
        return self._keys

    def __contains__(self, item):
        return EntityKey(item) in self._keys
        
```

This would make the syntax for dealing with the keys in the artifact class much simpler and separate the functionality for maintaining keyspace parity to a separate object.